dive_core/generatePm4Info_adreno.py

```diff
@@ -843,8 +850,20 @@ def outputPacketInfo(pm4_info_file, registers_et_root, enum_index_dict, opcode_d
     domain = registers_et_root.find('{http://nouveau.freedesktop.org/}domain[@name="'+packet_name+'"]')
     if domain is None:
       opcode = int(pm4_type_packet_value.attrib['value'],0)
-      pm4_info_file.write("    g_sPacketInfo[0x%x] = { \"%s\", 0, UINT8_MAX, {" % (opcode, packet_name))
-      pm4_info_file.write(" } };\n")
+
+      if 'variants' in pm4_type_packet_value.attrib:
+        variants = pm4_type_packet_value.attrib['variants']
+        variants_bitfield = GetGPUVariantsBitField(variants)
+        if (variants_bitfield != 0):
+          # kGPUVariantsBits has 6 bits
+          # it seems that the variant is only used for the non-domain ones
+          for i in range(6):
+            cur_variant_bitfield = (1<<i)
+            if cur_variant_bitfield & variants_bitfield:
+              pm4_info_file.write("    g_sPacketInfoVariant[(0x%x << kGPUVariantsBits) | 0x%x] = { \"%s\", 0, UINT8_MAX, {" % (opcode, cur_variant_bitfield, packet_name) + " } };\n")
+      else:
+        pm4_info_file.write("    g_sPacketInfo[0x%x] = { \"%s\", 0, UINT8_MAX, {" % (opcode, packet_name) + " } };\n")
```

Comment:
PEP 8 encourages using single-quoted strings here to avoid backslashes (https://peps.python.org/pep-0008/#string-quotes).

```suggestion
        pm4_info_file.write('    g_sPacketInfo[0x%x] = { "%s", 0, UINT8_MAX, {' % (opcode, packet_name) + " } };\n")
```

Reply:
thanks for pointing this. I will do a clean up in a different cl to make sure all formats are consistent (not just these ones in this cl)